### ISSUE TITLE
Automate Backups Used in Migration to a New Computer

### ISSUE THREAD

**Issue by wideaperture** [Zettlr/Zettlr#3712]
### Checklist

- [X] Zettlr does not possess this feature (if it does, but the feature doesn't work out for you, propose an enhancement instead.)
- [X] Zettlr's current features cannot be used to approximate this behaviour (otherwise, propose an enhancement instead.)
- [X] I have searched the documentation to make sure I did not overlook it.
- [X] I have searched both open and closed issues to see if someone has already proposed this.

### Feature Proposal

Having just completed the doc on the process for migrating Zettlr to a new computer, it seems to me that — long-term, at least — some aspects of this process could be automated. For example, Zettlr could automatically produce a tarball or zip file containing the relevant files from the application settings folder when a user selected `Export application settings` from a relevant menu or Preferences pane. Zettlr could similarly have a feature for importing these settings into a new installation.

Regarding workspaces and files, the locations of all of these are stowed in the backup of `config.json`. As I understand it now, when a new Zettlr install launches, possessing a backup of the previous install's `config.json` file, it checks the filepaths from the user's old machine and expunges them all when it discovers them to be dead links. A tweak to this process could be that, instead, Zettlr would prompt the user for the location of missing workspace folders. If the user has migrated these to the new machine, they could simply point Zettlr to the folders' new locations. To avoid a situation where the user is prompted 50 times for 50 different workspace folders, Zettlr could present them as a checklist with radio boxes next to each. The user could check only the ones they wanted to update, and the UI could include `Check all` and `Uncheck all` options.

### Example Workflow

Described above.

### Caveats

None that I can think of.

### Do you wish to attempt implementing this yourself?

No

### Platforms

- [X] Windows
- [X] macOS
- [X] Linux

**Comment by nathanlesage** [comment]
Just a few remarks:

> A tweak to this process could be that, instead, Zettlr would prompt the user for the location of missing workspace folders.

This actually already works somewhat: workspaces that cannot be found are marked "dead" with the option to re-scan them. Not exactly what you're proposing but it can serve as a reminder of which folders you have loaded.

And then: I think automating this is indeed a QoL improvement but since we already have a process description and there's a lot of other things on my plate this is low priority for now!

As always if someone wants to tackle it feel free ;)

**Comment by dustdfg** [comment]
If I just have a notes with links and without any workspaces, just copying folder should be fine or something else can go wrong?

**Comment by nathanlesage** [comment]
> If I just have a notes with links and without any workspaces, just copying folder should be fine or something else can go wrong?

Exactly. This issue is exclusively about configuration et al.; your actual notes and workspaces can be synced across devices without issues already.
